lib/manager/terraform/lockfile/hash.ts

```diff
@@ -0,0 +1,172 @@
+  const rootHash = crypto.createHash('sha256');
+
+  for (const file of files) {
+    // build for every file a line looking like "aaaaaaaaaaaaaaa  file.txt\n"
+    const hash = crypto.createHash('sha256');
+
+    // a sha256sum displayed as lowercase hex string to root hash
+    const fileBuffer = await fs.readFile(file);
+    hash.update(fileBuffer);
+    hash.end();
+    const data = hash.read();
+    rootHash.update(data.toString('hex'));
+
+    // add double space, the filename and a new line char
+    rootHash.update('  ');
+    const fileName = file.replace(/^.*[\\/]/, '');
+    rootHash.update(fileName);
+    rootHash.update('\n');
+  }
+
+  rootHash.end();
+  const rootData = rootHash.read();
+  const result: string = rootData.toString('base64');
+  return result;
+}
+
+export async function hashOfZipContent(
+  zipFilePath: string,
+  extractPath: string
+): Promise<string> {
+  await extract(zipFilePath, { dir: extractPath });
+  const files = await fs.readLocalDirectory(extractPath);
+  // the h1 hashing algorithms requires that the files are sorted by filename
+  const sortedFiles = files.sort((a, b) => a.localeCompare(b));
+  const filesWithPath = sortedFiles.map((file) => `${extractPath}/${file}`);
+
+  const result = hashFiles(filesWithPath);
+
+  // delete extracted files
+  for (const value of filesWithPath) {
+    await fs.deleteLocalFile(value);
+  }
+  return result;
+}
+
+async function getReleaseBackendIndex(
+  backendLookUpName: string,
+  version: string
+): Promise<VersionDetailResponse> {
+  return (
+    await http.getJson<VersionDetailResponse>(
+      `${defaultRegistryUrls[1]}/${backendLookUpName}/${version}/index.json`
+    )
+  ).body;
+}
+
+export async function calculateHashes(
+  builds: TerraformBuild[],
+  cacheDir: string
+): Promise<string[]> {
+  // for each build download ZIP, extract content and generate hash for all containing files
+  const hashes = await pMap(
+    builds,
+    async (build) => {
+      const downloadFileName = `${cacheDir}/${build.filename}`;
+      const extractPath = `${cacheDir}/extract/${build.filename}`;
+      logger.trace(
+        `Downloading archive and generating hash for ${build.name}-${build.version}...`
+      );
+      const readStream = http.stream(build.url);
+      const writeStream = fs.createWriteStream(downloadFileName);
+      readStream.pipe(writeStream);
+
+      const pipeline = fs.getStreamingPipeline();
+      try {
+        await pipeline(readStream, writeStream);
+      } catch (err) {
+        /* istanbul ignore next */
+        logger.error({ err }, 'write stream error');
+      }
+
+      const hash = await hashOfZipContent(downloadFileName, extractPath);
+      logger.trace(
+        { hash },
+        `Generated hash for ${build.name}-${build.version}`
+      );
+
+      // delete zip file
+      await fs.deleteLocalFile(downloadFileName);
+      await fs.deleteLocalFile(extractPath);
```

Comment:
```suggestion
      await fs.unlink(downloadFileName);
      await fs.remove(extractPath);
```

🤔 if i remember right, the local variants are for git repo relative files only. So you need to use normal `deleteFile`.

https://github.com/renovatebot/renovate/blob/460bfe95f030fcc1f0573d1bc17bb054c5b03cc5/lib/util/fs/index.ts#L55-L60

https://github.com/renovatebot/renovate/blob/095f2f8240ca90d3dc61016cb97b84d8e853c9b8/lib/util/fs/proxies.ts#L46-L53